packages/backend/src/extension.spec.ts

```diff
@@ -19,23 +19,17 @@
 import { beforeEach, expect, test, vi } from 'vitest';
 import type { ExtensionContext } from '@podman-desktop/api';
 import { activate, deactivate } from './extension';
+import { Studio } from './studio';
 
-const mocks = vi.hoisted(() => ({
-  studioActivateMock: vi.fn(),
-  studioDeactivateMock: vi.fn(),
-  studioConstructor: vi.fn(),
-}));
-
-vi.mock('./studio', () => ({
-  Studio: mocks.studioConstructor,
-}));
+vi.mock('./studio', () => {
+  const Studio = vi.fn();
+  Studio.prototype.activate = vi.fn();
+  Studio.prototype.deactivate = vi.fn();
+  return { Studio };
+});
 
 beforeEach(() => {
   vi.clearAllMocks();
```

Comment:
THe clear is weird? should be resetAllMock?

Reply:
as we don't redefine any mock, but only count the usage of them, `resetAllMock` is not strictly necessary